src/core/server/server.ts

```diff
@@ -88,18 +92,26 @@ export class Server {
     this.registerCoreContext(coreSetup);
     const pluginsSetup = await this.plugins.setup(coreSetup);
 
-    await this.legacy.setup({
+    const legacySetup = await this.legacy.setup({
       core: { ...coreSetup, plugins: pluginsSetup },
       plugins: mapToObject(pluginsSetup.contracts),
     });
 
+    await this.savedObjects.setup({
```

Comment:
> Kibana never listens for Http requests, not even with the "notReadyServer".

why? if Http service starts before Elasticsearch service

Reply:
I think I might have done something wrong when I tested this, when I tried again the notReadyServer shows on both this branch and master. I'll update the PR description.